test/common/buffer/owned_impl_test.cc

```diff
@@ -1256,6 +1256,13 @@ TEST_F(OwnedImplTest, MoveSmallSliceIntoNotEnoughFreeSpace) {
   TestBufferMove(4096 - 127, 128, 2);
 }
 
+TEST_F(OwnedImplTest, FrontSlice) {
+  Buffer::OwnedImpl buffer;
```

Comment:
Are there any tests that cover the case when there are leading zero length slices?